### PR TITLE
added tick labels from values demo

### PR DESCRIPTION
**Pull request by cammil**

### CODE REVIEW

**Comment by tacaswell** [comment]
Can you also add tweaking the locator to only use integers?

**Comment by cammil** [comment]
Done.

**Comment by tacaswell** [comment]
test failure is an intermittent one we have, kicked to re-start

File: examples/ticks_and_spines/tick_labels_from_values.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+fig = plt.figure()
+ax = fig.add_subplot(111)
+xs = range(26)
+ys = range(26)
+labels = list('abcdefghijklmnopqrstuvwxyz')
+
+
+def format_fn(tick_val, tick_pos):
+    try:
+        return labels[int(tick_val)]
```
Comment (tacaswell) [inline examples/ticks_and_spines/tick_labels_from_values.py]: Can you also check that the tick value is positive.  As it is if you pan to the negative values you get the alphabet in reverse!

Comment (cammil) [inline examples/ticks_and_spines/tick_labels_from_values.py]: I would be happy to do that, however, I personally feel that demos need to be as simple as possible. They should efficiently communicate just one idea. If I continue to put in various fixes, the thing we are trying to exemplify becomes lost. It makes it more difficult for people to learn from the demos. 

Any problems encountered after having learnt this idea could be treated as a personal programming problem, or perhaps found in another demo somewhere. This depends on the problems encountered.

In fact, I would argue that many matplotlib demos fall into this trap. There is too much other code which detracts from the main idea. 

If you still feel this demo would be better with the negative integers fixed, I will fix it.

Comment (WeatherGod) [inline examples/ticks_and_spines/tick_labels_from_values.py]: @cammil, you are absolutely right that many of matplotlib's demos fall into this trap, and there is ongoing work to address some of that. However, in this situation, I think the negative indices should be considered a bug, not a feature. The code will throw out any ticks greater than 'z', so panning one way gets one effect, while panning the other way gets another.

Maybe a compromise might be to use a modulus operator?

Comment (tacaswell) [inline examples/ticks_and_spines/tick_labels_from_values.py]: The counter point is that I think we should push the examples to be
something that users can copy-paste into their code to get the
functionality the example shows.  As @weathergod says, the negative ticks
getting would definitely be a bug and there is no reason to make everyone
who uses this fix it independently.

I am also thinking that maybe this should be a helper function included
someplace in the main namespace, just not sure where.

Comment (cammil) [inline examples/ticks_and_spines/tick_labels_from_values.py]: OK. I agree. I did not really think through how easy this fix would be. I will change it now...

**Comment by tacaswell** [comment]
Thanks!  This should show up http://matplotlib.org/devdocs/examples/ticks_and_spines/index.html in the near future.

**Pull request merged**